contracts/common/Registry.sol

```diff
@@ -68,6 +73,17 @@ contract Registry is Ownable {
     proofValidatorContracts[_validator] = true;
   }
 
+  function whitelistPredicate(address predicate, uint8 _type) public onlyOwner {
```

Comment:
add Type in params then we won't need branching of if else,
```   
   function whitelistPredicate(address predicate, Type _type) public onlyOwner {
    require(predicates[predicate]._type == Type.Invalid, "Predicate already added");	
      predicates[predicate]._type = _type;
    }
```

Reply:
Nice one, thanks. Changed it to this.